pkg/modprovider/planstore.go

```diff
@@ -156,6 +157,56 @@ func (e unknownAddressError) Error() string {
 	return fmt.Sprintf("unknown address: %q", e.addr)
 }
 
+// IsResourceDeleted returns true if the resource is deleted (not in the state).
+//
+// There are a couple different delete cases that this function handles:
+//  1. This is a `pulumi dn` operation and the entire stack is being deleted
+//     In this case there will be no `plan` because the `moduleStateHandler.Delete` method does not run plan.
+//  2. This is a `pulumi up` operation and this resource is being deleted or replaced.
+//     2a. If the resource is deleted (removed) then it won't be in the plan
+//     and we can check the state to know if it was successful
+//     2b. If the resource is replaced then the resource will remain in the state
+//     for both the success and failure case. We can't know for sure
+//
+// NOTE: This should only be called from within the `Delete` method of the child resource
+//
+// TODO[pulumi/pulumi-terraform-module#265] determine if the delete for this
+// specific resource succeeded
+func (s *planStore) IsResourceDeleted(
+	modUrn urn.URN,
+	addr ResourceAddress,
+) bool {
+	modState := s.getOrCreateStateEntry(modUrn).Await()
+	if planEntry, ok := s.getPlanEntry(modUrn); ok {
```

Comment:
This is still the minimal logic necessary to pass the tests right? That is it has to look at plan entries right?